lib/Adapters/RegressionAdapter/RegressionAdapter.js

```diff
@@ -0,0 +1,379 @@
+var logger = require('../../../common/logger')
+var validator = require('./hlsStreamRegressionValidator');
+var ErrorUtils = require('../../utils/error-utils');
+var glob = require('glob');
+
+var weblogger = logger.getLogger("RegressionAdapter-Web");
+var regressionAdapterConfig = config.get('regressionAdapter');
+var mediaServer = config.get('mediaServer');
+var entriesInfo = config.get('regressionAdapter').entries;
+var applicationName = mediaServer.applicationName;
+var hostname = mediaServer.hostname;
+var port = mediaServer.port;
+
+var testHelper;
+
+if (!regressionAdapterConfig || !regressionAdapterConfig.enable) {
+    return;
+}
+
+const master_playlist = {
+    '32':   '#EXTM3U\n\
+             #EXT-X-STREAM-INF:BANDWIDTH=5187246, URI="32/chunklist.m3u8", RESOLUTION=1280x720\n\
+             playlist.m3u8',
+    '33':   '#EXTM3U\n\
+             #EXT-X-STREAM-INF:BANDWIDTH=1094569, URI="33/chunklist.m3u8", RESOLUTION=480x270\n\
+             playlist.m3u8',
+    '34':   '#EXTM3U\n\
+             #EXT-X-STREAM-INF:BANDWIDTH=1447205, URI="34/chunklist.m3u8", RESOLUTION=640x360\n\
+             playlist.m3u8',
+    '35':   '#EXTM3U\n\
+             #EXT-X-STREAM-INF:BANDWIDTH=2205365, URI="35/chunklist.m3u8", RESOLUTION=640x360\n\
+             playlist.m3u8',
+};
+
+
+function readFile(res, full_path_file, entryId) {
+    fs.readFile(full_path_file, function (err, data) {
+        if (err) {
+            res.writeHead(404);
+            res.end(JSON.stringify(err));
+            testHelper.logger.error('[%s] failed to read path=%s. Error: %s', entryId, full_path_file, ErrorUtils.error2string(err));
+        }
+        res.writeHead(200);
+        res.end(data);
+    });
+}
+
+
+function EntryInfo(entryId, first_chunklist_index) {
+
+    this.id = entryId;
+    this.chunklist_index = {};
+    this.playlist = {};
+    this.initial_chunklist_index = first_chunklist_index;
+    this.last_chunklist_index = -1;
+    this.chunklist_count = 0;
+    this.data_warehouse_path = 'undefined';
+    this.flavors = {};
+}
+
+
+function TestHelper() {
+
+    this.rootFolderPath =  regressionAdapterConfig['dataWarehouseRootFolderPath'] ;
+    this.entries = {};
+    this.logger = logger.getLogger('regressionAdapter');
+    this.validators = {};
+}
+
+TestHelper.prototype.getFullLiveUrl  = function (entryId, flavor) {
+    var path = '/' + applicationName + "/_definst_/" + entryId + "_" + flavor + "/";
+    return url.format({
+        protocol : 'http',
+        hostname : hostname,
+        port : port,
+        pathname : path
+    });
+}
+
+TestHelper.prototype.initEntries = function() {
+
+    var that = this;
+    var root_mp4_path = config.get('rootFolderPath');
+    var p;
+    p = _.map(entriesInfo, function (entryConfig) {
+        // remove directory
+        var fullpath = root_mp4_path + '/' + entryConfig.entryId;
+        return fs_utils.cleanFolder(fullpath, 'obsolete').then(function () {
```

Comment:
cleanFolder of fs_utils is not really cleaning the folder, but it only renames it to another location (performance)
suggest you use https://www.npmjs.com/package/q-io#removetreepath